### PR TITLE
fix: maintain scroll position when navigating back to main page

### PR DESCRIPTION
**Pull request by jthistle**
https://trello.com/c/ci2fkdj7/93-maintain-scroll-position-vv-e

### CODE REVIEW

**Comment by marxian** [comment]
#### 📝Changed routes:


| [**/**](https://my-best-life-lx07z1443-neontribe.vercel.app/) |
|:-:|
|<a href="https://my-best-life-lx07z1443-neontribe.vercel.app/">Screenshot of /</a><br /><sup><a href="https://images.weserv.nl?url=https://s.microlink.io/sSHrbetZnbt9qYQnsekhgSLRY3M_.png">(view full size)</a> |







Commit 09cb90db3e621c551404c3853e4476f207352c8f (https://my-best-life-lx07z1443-neontribe.vercel.app).

File: src/hooks/scrollRemember.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  resumePosition(): void,
+}
+
+export const useScrollRemember = (): UseScrollRememberReturns => {
+  const router = useRouter()
+
+  // Store last scroll position so we can return to it when we load the page next time
+  useEffect(() => {
+    const handler = () => {
+      localStorage.setItem('lastScroll/' + router.pathname, window.pageYOffset.toString())
```
Comment (edpe) [inline src/hooks/scrollRemember.tsx:14]: Does this list position persist even if I navigate away from the app and come back to it, and is this the behaviour we want? What would be the effects of using session storage or sharing state between page navigations?

**Review by edpe — CHANGES_REQUESTED** [review]

File: src/hooks/scrollRemember.tsx (hunk 1 above)
Comment (jthistle) [reply src/hooks/scrollRemember.tsx:14]: Yes, this would happen. I can try to hook onto some 'leavePage' event and clear it, maybe?

**Review by edpe — APPROVED** [review]
It works for me :+1: 

It still seems to remember previous scroll position but I just discovered that Chrome remembers scroll position anyway when you nav away and then return to the page from another domain, so maybe it was fine anyway? Firefox and Safari don't apparently. Either way it's working when you go back to the list from the card which is what we want.

**Comment by jthistle** [comment]
@edpe that's interesting, it wasn't working for me in Chromium 84. Perhaps it's a recent feature. Thanks for the review.

**Pull request merged**